File: app/src/main/java/com/example/kotlinbuttontest/ApiService.java
package com.example.kotlinbuttontest;

import com.google.gson.JsonObject;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.internal.http.HttpHeaders;
import retrofit2.Call;
import retrofit2.http.*;


public interface ApiService {

    @Multipart
    @POST("/users")
    Call<ResponseBody> testcall(@Query("name") String name, @Query("part") String part, @Part MultipartBody.Part image);

    @POST("/users")
    Call<ResponseBody> test();

    @GET("/users")
    Call<JsonObject> getUser(@Query("name") String name);

    @GET("/users")
    Call<JsonObject> getTotalUser();
}

File: app/src/main/java/com/example/kotlinbuttontest/apiService.kt
package com.example.kotlinbuttontest

import com.google.gson.JsonObject
import com.google.gson.JsonPrimitive
import io.reactivex.Observable
import io.reactivex.Single
import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

interface apiService {
    @Multipart
    @POST("/users")
    fun testcall(@Query("name") name: String?, @Query("part") part: String?, @Part image: MultipartBody.Part?): Call<ResponseBody>

    @POST("/users")
    fun test(): Call<ResponseBody?>?

    @GET("/users")
    fun getUser(@Query("name") name: String?): Observable<JsonObject>

    @GET("/users")
    fun getTotalUser(): Observable<JsonObject>

    @GET
    @Streaming
    fun getImageDetails(@Url url : String) : Call<ResponseBody>
}